documentation/src/docs/recovery-procedures/table-segment-recover-pre-index-full.md

```diff
@@ -0,0 +1,96 @@
+- `tables.systemcritical.unindexed.bytes.max` (_defaults to 256MB_): Same as the previous parameter, but for Metadata
+Table Segments (i.e., `container_metadata` and  `storage_metadata` on each Segment Container).
+
+In rare cases in which LTS is super slow or unavailable for hours, but there is some activity happening in a Hash-based
+Table Segment, we can find errors in the logs such as:
+
+```
+2022-08-08 13:07:57,194 1192230051 [core-19] INFO i.p.s.s.tables.ContainerKeyIndex - KeyIndex[27]: Table Segment 14 cannot perform tail-caching because tail index too long (150542984).
+```
+```
+2022-08-08 13:07:57,194 1192230051 [core-19] INFO i.p.s.s.tables.ContainerKeyIndex - KeyIndex[27]:: System-critical TableSegment 3 is blocked due to reaching max unindexed size.
+```
+
+Messages like the ones above indicate that Table Segments may not be able to recover, to write new entries, or both.
+If the impacted Hash-based Table Segments are critical for the operation of Pravega, it may leave the impacted Segment
+Container unable for processing new operations. Once this situation is reached, recovering the Segment Container requires 
+manual intervention.
+
+# Repair Procedure
+
+The repair procedure is relatively simple and works as follows:
+
+1. _Make sure that LTS is available and works properly_: The root cause of the problem is related to LTS not being able
+to store data fast enough, or just being unavailable for a very long period of time. Any attempt to recover Pravega
+first requires to be sure that LTS is working fine.
+
+
+2. _Temporarily update Table Segment un-indexed data parameters_: Once LTS is back to working state, we need to increase
+the values in the aforementioned parameters. Concretely, the following errors related to the configuration parameters:
+i) `...tail index too long.` requires increasing `tables.preindex.bytes.max`, 
+ii) `...recovery timed out.` requires increasing `tables.recovery.timeout.millis`, 
+iii) `...blocked due to reaching max unindexed size` requires increasing `tables.unindexed.bytes.max` or
+`tables.systemcritical.unindexed.bytes.max`, depending on whether the Table Segment is a metadata one (i.e., 
+`container_metadata` and `storage_metadata` on each Segment Container) or not.
+The values to be set should be enough to get through the blocker issue. For instance, if one Table Segment shows a log
+like `cannot perform tail-caching because tail index too long (150542984)`, it is reasonable to configure
+`tables.preindex.bytes.max` to 256MBs. However, setting an excessively high value may be problematic in the case that
+LTS is still not working properly, as data may continue getting accumulated up to the configured value.
+If multiple error messages are found related to Table Segments, you can update all these parameters to higher values.
+Also, note that if the "pre-index" or "tail-caching" process involves storing data in cache. For this reason, in some 
```

Comment:
Correctness: "note that if" -> "note that"

Cf.,
> 'Also, note that **since** the "pre-index" or "tail-caching" process involves storing data in cache, it may be necessary to also temporarily increase cache size...'